### PR TITLE
Access Rules update

### PR DESCRIPTION
**Pull request by heylemur**
Summary of changes:
- Removed the granting access examples from the Managing_Accounts folders
- Deleted the Composite_Roles examples
- Added tags to the Managing_Resources examples
- Replaced the Managing_Roles examples with the new
- Added info on migrating to access rules to the Readme files

### CODE REVIEW

**Review by turner-strongdm — APPROVED** [review]
Spoke with team. g2g.

File: README.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ This is the examples repository for the [strongDM Ruby SDK](https://github.com/s
 
 ---
 > **NOTE:**  
-> This repository is currently under construction. The content is expected to change in the coming weeks.
-
+> To increase flexibility when managing thousands of Resources, Role Grants have
```
Comment (heylemur) [inline README.md:7]:
```suggestion
> To increase flexibility when managing a large volume of Resources, Role Grants have
```

File: README.md [hunk 2]
```diff
@@ -4,6 +4,12 @@ This is the examples repository for the [strongDM Ruby SDK](https://github.com/s
...
+been deprecated in favor of Access Rules, which allow you to grant access based
+on Resource Tags and Type.
```
Comment (heylemur) [inline README.md:9]:
```suggestion
on Resource tags and type.
```

File: README.md [hunk 3]
```diff
@@ -4,6 +4,12 @@ This is the examples repository for the [strongDM Ruby SDK](https://github.com/s
...
+>
+> Previously, you would grant a Role access to specific resources by ID via Role
```
Comment (heylemur) [inline README.md:11]:
```suggestion
> Previously, you would grant a Role access to specific Resources by ID via Role
```

File: README.md [hunk 4]
```diff
@@ -4,6 +4,12 @@ This is the examples repository for the [strongDM Ruby SDK](https://github.com/s
...
+Grants. Now, when using Access Rules, the best practice is to grant Resources access based on Type and Tags.
```
Comment (heylemur) [inline README.md:12]:
```suggestion
Grants. Now, when using Access Rules, the best practice is to give Roles access to Resources based on type and tags.
```

File: README.md [hunk 5]
```diff
@@ -4,6 +4,12 @@ This is the examples repository for the [strongDM Ruby SDK](https://github.com/s
...
+>
+>The following examples demonstrate Dynamic Access Rules with Tags and Resource Types, as well as Static Access Rules for backwards compatibility with Role Grants. If it is _necessary_ to grant access to specific Resources in the same way as RoleGrants did, you can use Resource IDs directly in Access Rules.
```
Comment (heylemur) [inline README.md:14]:
```suggestion
>The following examples demonstrate Dynamic Access Rules with tags and Resource types, as well as Static Access Rules. If it is _necessary_ to grant access to specific Resources in the same way as RoleGrants did, you can use Resource IDs directly in Access Rules.
```

File: 3_managing_roles/grant_access.rb [hunk 6]
```diff
@@ -12,89 +12,99 @@
...
+  $client.roles.update(role).role
+end
+
+def main
+  # Each of the following functions is an example of how to do an operation using Access Rules.
+
+  create_and_update_access_rules
+
+ 	# The RoleGrants API has been deprecated in favor of Access Rules.
+  # When using Access Rules, the best practice is to grant Resources access based on Type and Tags.
```
Comment (heylemur) [inline 3_managing_roles/grant_access.rb:101]:
```suggestion
 	# When using Access Rules, the best practice is to grant Resources access based on type and tags.
```

File: 3_managing_roles/grant_access.rb [hunk 7]
```diff
@@ -12,89 +12,99 @@
...
+	# If it is _necessary_ to grant access to specific Resources in the same way as RoleGrants did,
```
Comment (heylemur) [inline 3_managing_roles/grant_access.rb:102]:
```suggestion
	# If it is _necessary_ to grant access to specific Resources in the same way as Role Grants did,
```

File: 3_managing_roles/grant_access.rb [hunk 8]
```diff
@@ -12,89 +12,99 @@
...
+
+  $client.roles.update(role).role
+end
+
+def main
+  # Each of the following functions is an example of how to do an operation using Access Rules.
+
+  create_and_update_access_rules
+
+ 	# The RoleGrants API has been deprecated in favor of Access Rules.
```
Comment (heylemur) [inline 3_managing_roles/grant_access.rb:100]:
```suggestion
 	# The Role Grants API has been deprecated in favor of Access Rules.
```

File: 3_managing_roles/grant_access.rb [hunk 9]
```diff
@@ -12,89 +12,99 @@
...
+  # When using Access Rules, the best practice is to grant Resources access based on Type and Tags.
+	# If it is _necessary_ to grant access to specific Resources in the same way as RoleGrants did,
+	# you can use Resource IDs directly in Access Rules as shown in the following examples.
+
+	create_role_grant_via_access_rules
```
Comment (heylemur) [inline 3_managing_roles/grant_access.rb:105]:
```suggestion
 create_role_grant_via_access_rules
```

File: 3_managing_roles/revoke_access.rb [hunk 10]
```diff
@@ -12,92 +12,99 @@
...
+  $client.roles.update(role).role
+end
+
+def main
+  # Each of the following functions is an example of how to do an operation using Access Rules.
+
+  create_and_update_access_rules
+
+ 	# The RoleGrants API has been deprecated in favor of Access Rules.
+  # When using Access Rules, the best practice is to grant Resources access based on Type and Tags.
```
Comment (heylemur) [inline 3_managing_roles/revoke_access.rb:101]:
```suggestion
 	# When using Access Rules, the best practice is to grant Resources access based on type and tags.
```

File: 3_managing_roles/revoke_access.rb [hunk 11]
```diff
@@ -12,92 +12,99 @@
...
+	# If it is _necessary_ to grant access to specific Resources in the same way as RoleGrants did,
+	# you can use Resource IDs directly in Access Rules as shown in the following examples.
+
+	create_role_grant_via_access_rules
```
Comment (heylemur) [inline 3_managing_roles/revoke_access.rb:105]:
```suggestion
create_role_grant_via_access_rules
```

File: 3_managing_roles/revoke_access.rb [hunk 12]
```diff
@@ -12,92 +12,99 @@
...
+end
+
+def main
+  # Each of the following functions is an example of how to do an operation using Access Rules.
+
+  create_and_update_access_rules
+
+ 	# The RoleGrants API has been deprecated in favor of Access Rules.
+  # When using Access Rules, the best practice is to grant Resources access based on Type and Tags.
+	# If it is _necessary_ to grant access to specific Resources in the same way as RoleGrants did,
```
Comment (heylemur) [inline 3_managing_roles/revoke_access.rb:102]:
```suggestion
	# If it is _necessary_ to grant access to specific Resources in the same way as Role Grants did,
```

File: 3_managing_roles/update_role.rb [hunk 13]
```diff
@@ -12,45 +12,99 @@
...
+  $client.roles.update(role).role
+end
+
+def main
+  # Each of the following functions is an example of how to do an operation using Access Rules.
+
+  create_and_update_access_rules
+
+ 	# The RoleGrants API has been deprecated in favor of Access Rules.
+  # When using Access Rules, the best practice is to grant Resources access based on Type and Tags.
```
Comment (heylemur) [inline 3_managing_roles/update_role.rb:101]:
```suggestion
 	# When using Access Rules, the best practice is to grant Resources access based on type and tags.
```

File: 1_managing_resources/delete_resource.rb [hunk 14]
```diff
@@ -31,25 +31,26 @@
...
+# Define a Postgres Datasource
 postgres = SDM::Postgres.new(
   name: 'Example Postgres Datasource',
   hostname: 'example.strongdm.com',
   port: 5432,
   username: 'example',
   password: 'example',
   database: 'example',
-  port_override: 19_999
+  port_override: 19_999.
```
Comment (heylemur) [inline 1_managing_resources/delete_resource.rb:42]:
```suggestion
  port_override: 19_999,
```

File: 1_managing_resources/create_datasource.rb [hunk 15]
```diff
@@ -39,10 +39,11 @@
   username: 'example',
   password: 'example',
   database: 'example',
-  port_override: 19_999
+  port_override: 19_999,
+  tags: '{"env": "example"}'
```
Comment (heylemur) [inline 1_managing_resources/create_datasource.rb:43]:
```suggestion
  tags: '{"env": "example"}',
```

**Review by jeffreylees — APPROVED** [review]

**Pull request merged**